Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -142,19 +142,30 @@ void ACesium3DTileset::SampleHeightMostDetailed(
         position.Z));
   }
 
+  size_t count = positions.size();
+
   CesiumAsync::Future<Cesium3DTilesSelection::SampleHeightResult> future =
       this->_pTileset
           ? this->_pTileset->sampleHeightMostDetailed(positions)
           : getAsyncSystem().createResolvedFuture(
                 Cesium3DTilesSelection::SampleHeightResult{
                     std::move(positions),
-                    std::vector<bool>(positions.size(), false),
+                    std::vector<bool>(count, false),
                     {"Could not sample heights from tileset because it has not "
                      "been created."}});
 
-  std::move(future).thenImmediately(
-      [this, OnHeightsSampled = std::move(OnHeightsSampled)](
-          Cesium3DTilesSelection::SampleHeightResult&& result) {
+  std::move(future)
+      .catchImmediately([positions = std::move(positions)](
```

Comment:
Checking my understanding: will this only trigger if `pTileset` is valid? (and therefore `std::move(positions)` isn't done twice if `pTileset == nullptr` ?)

Reply:
Hmm yeah it will be fine, though it is a little sloppy. As you said, the first std::move only happens if the _pTileset is nullptr. The second, capturing for the `catchImmediately`, happens in all cases, so technically we're moving `positions` twice. But the it's harmless because a `createResolvedFuture` will never result in a rejected future, and so that catchImmediately's continuation will never be called in the case of a null tileset.

I'll see if I can reorganize this a little to be less sloppy and to require less analysis to be sure it's safe.